viewing_party/party.py

```diff
@@ -0,0 +1,169 @@
+    if list_ratings:
+        average_result =  average(list_ratings)
+    return average_result
+
+
+def get_most_watched_genre(user_data):
+    popular_genre = None
+    if user_data["watched"]: 
+        outer_key = "watched"
+        inner_key = "genre"
+        list_genre = get_dict_value(user_data, outer_key, inner_key)
+        popular_genre = most_frequent(list_genre)
+    return popular_genre
+
+'''
+    wave 3
+'''
+
+def get_unique_watched(user_data):
+    user_watched_list = []
+    friends_watched_list =[]
+    create_user_watched_list(user_data, user_watched_list)
+    create_friends_watched_list(user_data,friends_watched_list)  
+    unique_in_user = set(user_watched_list).difference(set(friends_watched_list))
+    result =  [{"title":item} for item in unique_in_user]
+    return result
+
+
+def get_friends_unique_watched(user_data):
+    user_watched_list = []
+    friends_watched_list =[]
+    create_user_watched_list(user_data, user_watched_list)
+    create_friends_watched_list(user_data,friends_watched_list)  
+    unique_in_friends = set(friends_watched_list).difference(set(user_watched_list))
+    result =  [{"title":item} for item in unique_in_friends]
+    return result
+
+'''
+    wave 4
+'''
+
+def get_available_recs(user_data):
+    friends_list = []
+    recommendations = []
+    user_list = [item['title'] for item in user_data['watched']]
+    host_list = user_data['subscriptions']
+    create_friends_list(user_data,friends_list) 
+    for i in friends_list:  
+        if (i['title'] not in user_list and i['host'] in host_list) and (i not in recommendations):  
+            recommendations.append(i)
+        elif (not user_list and i['host'] in host_list) and (i not in recommendations): 
+            recommendations.append(i)
+    return recommendations
+
+'''
+    wave 5
+'''
+
+
+def get_new_rec_by_genre(user_data):
+    friends_list = []
+    genre_recom = []
+    user_list_genre= [item['genre'] for item in user_data['watched']] #list of dictionaries
+    user_list_title = [item['title'] for item in user_data['watched']] #list of dictionaries  
+    create_friends_list(user_data,friends_list)
+    for i in friends_list:  
+        if i['title'] not in user_list_title and i['genre'] in user_list_genre: 
+            if i not in genre_recom:
+                genre_recom.append(i)  
+    return genre_recom
+
+
+def get_rec_from_favorites(user_data):
+    favorites = [item['title'] for item in user_data["favorites"]]
+    friends_list = [] 
+    create_friends_watched_list(user_data,friends_list)     
+    not_watched = set(favorites).difference(set(friends_list))
+    result= [{"title":item} for item in not_watched]
+    return result
+
+''' 
+    helper functions
+'''
+def search_value(user_data,movie,search_for):
+    if movie not in user_data[search_for]:
+        user_data[search_for] += [movie]
+    return user_data
+
+
+def most_frequent(list_value):
+    count = Counter(list_value)
```

Comment:
Good work researching this library. It would be a useful exercise to build a frequency map yourself: https://www.geeksforgeeks.org/counting-the-frequencies-in-a-list-using-dictionary-in-python/ .